lib/jekyll/admin/api_handler.rb

```diff
@@ -0,0 +1,30 @@
+module Jekyll
+  module Admin
+    class ApiHandler
+      def initialize(site)
+        @site = site
+      end
+
+      def read_file(path)
+        File.read(path)
+      end
```

Comment:
For `_config.yml`, you may need to access the raw file to get it without Jekyll's defaults (which there may be a Jekyll helper method for), but for the most part, for pages, data, etc. we should be pulling it from the site object, not reading it from disk and re-parsing it ourselves... let Jekyll do the heavy lifting for consistency.